dbms/src/Debug/dbgKVStore/dbgFuncInvestigator.cpp

```diff
@@ -0,0 +1,287 @@
+#include <Interpreters/InterpreterCreateQuery.h>
+#include <Interpreters/InterpreterSelectQuery.h>
+#include <Parsers/ASTIdentifier.h>
+#include <Parsers/ASTLiteral.h>
+#include <Parsers/ASTSelectQuery.h>
+#include <Parsers/ParserCreateQuery.h>
+#include <Parsers/ParserSelectQuery.h>
+#include <Parsers/parseQuery.h>
+#include <Storages/IManageableStorage.h>
+#include <Storages/KVStore/KVStore.h>
+#include <Storages/KVStore/MultiRaft/RegionData.h>
+#include <Storages/KVStore/MultiRaft/RegionRangeKeys.h>
+#include <Storages/KVStore/Region.h>
+#include <Storages/KVStore/TMTContext.h>
+#include <TiDB/Schema/TiDBSchemaManager.h>
+
+#include <unordered_map>
+#include <unordered_set>
+
+namespace DB
+{
+
+struct MatchResult
+{
+    String mapped_database_name;
+    String table_name;
+    std::vector<std::pair<UInt64, UInt64>> in_default;
+    std::vector<std::pair<UInt64, UInt64>> in_write;
+    std::vector<UInt64> in_lock;
+    std::unordered_map<UInt64, RegionPtr> regions;
+
+    std::string toString() const
+    {
+        FmtBuffer fmt_buf;
+        fmt_buf.fmtAppend("default_cf ");
+        fmt_buf.joinStr(
+            in_default.begin(),
+            in_default.end(),
+            [](const auto & a, const auto &) { return fmt::format("{}-{}", a.first, a.second); },
+            ":");
+        fmt_buf.fmtAppend("; write_cf ");
+        fmt_buf.joinStr(
+            in_write.begin(),
+            in_write.end(),
+            [](const auto & a, const auto &) { return fmt::format("{}-{}", a.first, a.second); },
+            ":");
+        fmt_buf.fmtAppend("; lock_cf ");
+        fmt_buf.joinStr(in_lock.begin(), in_lock.end(), ":");
+        for (const auto & [region_id, region] : regions)
+        {
+            fmt_buf.fmtAppend(
+                "; region {} {}, tikv_range: {}; ",
+                region_id,
+                region->getDebugString(),
+                region->getRange()->toDebugString());
+        }
+        return fmt_buf.toString();
+    }
+};
+
+void dbgFuncFindKey(Context & context, const ASTs & args, DBGInvoker::Printer output)
+{
+    if (args.size() < 3)
+        throw Exception(
+            "Args not matched, should be: database-name, table-name, start1 [, start2, ..., end1, end2, ...]",
+            ErrorCodes::BAD_ARGUMENTS);
+
+    auto & tmt = context.getTMTContext();
+    auto & kvstore = *tmt.getKVStore();
+    MatchResult result;
+    const String & database_name_raw = typeid_cast<const ASTIdentifier &>(*args[0]).name;
+    const String & table_name = typeid_cast<const ASTIdentifier &>(*args[1]).name;
+
+    auto maybe_database_name = mappedDatabaseWithOptional(context, database_name_raw);
+    if (maybe_database_name == std::nullopt)
+    {
+        output(fmt::format("Database {} not found.", database_name_raw));
+        return;
+    }
+    result.mapped_database_name = maybe_database_name.value();
+    auto & mapped_database_name = result.mapped_database_name;
+    result.table_name = table_name;
```

Comment:
I think we also need to map the `table_name` in tidb to get its name in TiFlash (which is `t_${table_id}`)?

Reply:
Seems `tmt.getStorages().getByName(mapped_database_name, table_name, false)` here takes the un-mapped name, which is not `t_$tid`, but its user name.